### PR TITLE
[SPARK-39289][CORE][SQL][SS] Replace `map(_.toBoolean).getOrElse(false/true)` with `exists/forall(_.toBoolean)`

### PR DESCRIPTION
**Pull request by dcoliversun**
### What changes were proposed in this pull request?

This PR has two change
* replace `map(_.toBoolean).getOrElse(false)` with `exists(_.toBoolean)`. 
> `exists` in Scala refer to https://github.com/scala/scala/blob/c850a83a72e6514cc7e55b2c1bf01e27c0e028a3/src/library/scala/Option.scala#L362-L376
* replace `map(_.toBoolean).getOrElse(true)` with `forall(_.toBoolean)`
> `forall` in Scala refer to https://github.com/scala/scala/blob/c850a83a72e6514cc7e55b2c1bf01e27c0e028a3/src/library/scala/Option.scala#L378-L390

### Why are the changes needed?

Code Simplification.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

No need to add unit test for this code simplification.

### CODE REVIEW

**Review by srowen — COMMENTED** [review]
I dont' mind the change, though it won't have any impact in these usages. Are there more? esp. those where the list is not just 0 or 1 elements?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala [hunk 1]
```diff
@@ -54,31 +54,31 @@ private[sql] class JSONOptions(
   val samplingRatio =
     parameters.get("samplingRatio").map(_.toDouble).getOrElse(1.0)
   val primitivesAsString =
-    parameters.get("primitivesAsString").map(_.toBoolean).getOrElse(false)
+    parameters.get("primitivesAsString").exists(_.toBoolean)
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala:57]: I think this case actually map and getOrElse is easier to read ..

Comment (srowen) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala:57]: I'm OK with that too. It's more performant if it were a list, but slightly less readable, and this is an Option

Comment (dcoliversun) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala:57]: It's worth sacrificing performance for readability. If we agree that `exists/forall` is not easier to read, I will close this PR and keep it as it is.

Comment (srowen) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala:57]: Yeah I think we might just leave it as is

Comment (dcoliversun) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala:57]: OK. I close this PR. Thanks! @srowen @HyukjinKwon

**Pull request closed without merging**